src/Common/MetricHelper.cs

```diff
@@ -481,17 +482,27 @@ public void FlushMetric()
                 return;
             }
 
-            try
+            FlushMetricAsync(TelemetryClients, CancellationToken.None);
+        }
+
+        private Task FlushMetricAsync(IEnumerable<TelemetryClient> TelemetryClients, CancellationToken cancellationToken)
+        {
+            foreach (TelemetryClient client in TelemetryClients)
             {
-                foreach (TelemetryClient client in TelemetryClients)
+                if(cancellationToken != CancellationToken.None && cancellationToken.IsCancellationRequested)
+                {
+                    return Task.FromCanceled(cancellationToken);
+                }
+                try
                 {
                     client.Flush();
```

Comment:
We should call the async version of the API, otherwise it's still running synchronously.
```csharp
client.FlushAsync();
```

Reply:
Async issue is fixed. Please review.